multicast/recv.py

```diff
@@ -146,21 +269,49 @@ def parseArgs(arguments=None):
 	return parser.parse_args(arguments)
 
 
-def run(groups, port, iface=None, bind_group=None):
-	"""The work-horse function. Spawns a listener for multicast based on arguments.
-	generally speaking you want to bind to one of the groups you joined in
-	this module/instance, but it is also possible to bind to group which
-	is added by some other programs (like another python program instance of this)
-	"""
- 
- 	# assert bind_group in groups + [None], \
-	#     'bind group not in groups to join'
-	sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+def hearstep(groups, port, iface=None, bind_group=None):
+	"""Will listen on the given port of an interface for multicast messages to the given group(s).
 
-	# allow reuse of socket (to allow another instance of python running this
-	# script binding to the same ip/port)
-	sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-	
+	The work-horse function.
+
+
+	Minimal Acceptance Testing:
+
+	First setup test fixtures by importing multicast.
+
+		>>> import multicast
+		>>> multicast.recv is not None
+		True
+		>>>
+
+	Testcase 1: Stability testing.
+
+		>>> import multicast
+		>>>
+		>>> multicast.recv is None
+		False
+		>>>
+		>>> multicast.recv.hearstep is None
+		False
+		>>> type(multicast.recv.hearstep)
+		<class 'function'>
+		>>> multicast.recv.hearstep(None, 19991) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<BLANKLINE>
+		'...'
+		>>> multicast.recv.hearstep([multicast.__MCAST_DEFAULT_GROUP], 19991) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
+		<BLANKLINE>
+		'...'
+		>>> multicast.recv.hearstep([multicast.__MCAST_DEFAULT_GROUP], 19991, None, multicast.__MCAST_DEFAULT_GROUP) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
```

Comment:
🙅🏻 Linter is right ; lines are too LONG
```suggestion
		'...'
		>>> tst_fixture = multicast.__MCAST_DEFAULT_GROUP
		>>> multicast.recv.hearstep([tst_fixture], 19991) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
		<BLANKLINE>
		'...'
		>>> multicast.recv.hearstep([tst_fixture], 19991, None, tst_fixture) #doctest: -DONT_ACCEPT_BLANKLINE, +ELLIPSIS
```